t/api/08-jobtemplates.t

```diff
@@ -262,6 +263,83 @@ is_deeply(
     "Initial job templates"
 ) || diag explain $t->tx->res->json;
 
+subtest 'to_template' => sub {
+    my $yaml1 = <<'EOM';
+defaults:
+  i586:
+    machine: 64bit
+    priority: 50
+products:
+  opensuse-13.1-DVD-i586:
+    distri: opensuse
+    flavor: DVD
+    version: '13.1'
+scenarios:
+  i586:
+    opensuse-13.1-DVD-i586:
+    - textmode:
+        machine: 32bit
+        priority: 40
+    - textmode:
+        machine: 64bit
+        priority: 40
+    - kde:
+        priority: 40
+    - client1:
+        machine: 32bit
+        priority: 40
+    - client1:
+        machine: 64bit
+        priority: 40
+    - server:
+        machine: 32bit
+        priority: 40
+    - server:
+        machine: 64bit
+        priority: 40
+    - client2:
+        machine: 64bit
+        priority: 40
+    - client2:
+        machine: 32bit
+        priority: 40
+    - advanced_kde:
+        priority: 40
+        settings:
+          ADVANCED: '1'
+          DESKTOP: advanced_kde
+EOM
+    my $yaml2 = <<'EOM';
+products: {}
+scenarios: {}
+EOM
+    my %yaml = (1001 => $yaml1, 1002 => $yaml2);
```

Comment:
1. Reading this test I'm thinking if there shouldn't be a test for a group both with and without YAML, since `to_yaml` has to handle that.
2. It also occurs to me you can't tell from the test if that's already the case. Would you mind adding comments that explain why you picked those groups?
3. The name `to_template` seems wrong given that it tests `to_yaml`.

Reply:
1. probably. I think with the changes more tests can be refactored, but I simply didn't have the time
2. I tested only those job groups that are already in the database at that time
3. will fix